src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/TableSchemaResolver.java

```diff
@@ -104,4 +138,32 @@ private String inferDataTypeFromJsonObject(JsonNode value) {
     // Passing null to schemaName when there is no schema information
     return columnTypeMapper.mapToColumnType(schemaType);
   }
+
+  private static class ColumnValuePair {
+    private final String columnName;
+    private final String quotedColumnName;
+    private final JsonNode jsonNode;
+
+    public static ColumnValuePair of(Map.Entry<String, JsonNode> field) {
```

Comment:
`of` suggests we construct the object with a given value. However, here, we apply some conversion - extracting fields, so `from` would be more suitable.